### PR TITLE
Move core and key to kv package 

### PR DESCRIPTION
**Pull request by tensorchen**
See #704 

This relates to #261

### CODE REVIEW

**Comment by tensorchen** [comment]
This is the last step of core package refactoring. In this PR, core package will be completely eliminated.

File: api/kv/key.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+import (
+	"go.opentelemetry.io/otel/api/kv/value"
+)
+
+// Key represents the key part in key-value pairs. It's a string. The
+// allowed character set in the key depends on the use of the key.
+type Key string
+
+// NewKey creates a new key with a passed name.
```
Comment (jmacd) [inline api/kv/key.go:25]: I wonder if we can now just remove this function. The caller can use `kv.Key(...)` instead of NewKey, that reads better to me.

Comment (jmacd) [reply api/kv/key.go:25]: Since we'd like to merge this ASAP, I will fix this.

File: api/kv/doc.go [hunk 2]
```diff
@@ -17,4 +17,4 @@
 //
```
Comment (Aneurysm9) [inline api/kv/doc.go:17]: The copied documentation above this line is no longer correct.  This package doesn't provide "values, numbers[, or] span contexts".

**Comment by jmacd** [comment]
I will follow immediately with a removal for NewKey. This change is difficult to work with.

**Pull request merged**